CodeHawk/README.md

```diff
@@ -8,166 +8,39 @@ Building codehawk requires the following applications and libraries:
 - The Findlib / ocamlfind library manager
 - The Zlib C library, version 1.1.3 or up
 - The Zarith library
+- goblint-cil, version 2.0.6
 
 The CodeHawk Tool Suite contains three analyzers that can all be built
 individually. All three analyzers have an optional gui that can be built
 separately (and that requires additional dependencies to be installed).
+The gui is current out-of-date and will require additional work to revive.
 
 
-### Install dependencies: Ubuntu 16.04 or later (without gui)
-
-```
-sudo apt update -y
-sudo apt install software-properties-common pkg-config m4 zlib1g-dev libgmp-dev -y
-sudo add-apt-repository ppa:avsm/ppa -y
-sudo apt update -y
-sudo apt install opam
-git clone https://github.com/static-analysis-engineering/codehawk.git
-opam init --bare --no-setup --disable-sandboxing
-opam switch create codehawk-4.12.1 4.12.1 --no-switch
-eval $(opam env --switch=codehawk-4.12.1 --set-switch)
-opam install ocamlfind zarith camlzip extlib goblint-cil
-cd codehawk/CodeHawk
-```
-
-Depending on which analyzer you want to build:
-- **Binary:** `./make_binary_analyzer.sh`
-  - analyzer binary: CHB/bchcmdline/chx86_analyze
-- **C:** `./make_c_analyzer.sh`
-  - parser binary: CHC/cchcil/parseFile
-  - analyzer binary: CHC/cchcmdline/canalyzer
-- **Java:** `./make_java_analyzer.sh`
-  - analyzer binary: CHJ/jchstac/chj_initialize
-- **all:** `./full_make_no_gui.sh`
-  - analyzer/parser binaries: all of the above
+### Install Dependencies:
 
-
-### Install dependencies and build: Ubuntu 16.04 or later (including gui)
+These commands should work for Ubuntu 22.04+:
 
 ```
 sudo apt update -y
-sudo apt install software-properties-common pkg-config m4 zlib1g-dev libgmp-dev liblablgtk2-ocaml-dev liblablgtksourceview2-ocaml-dev -y
-sudo add-apt-repository ppa:avsm/ppa -y
-sudo apt update -y
-sudo apt install opam
-git clone https://github.com/static-analysis-engineering/codehawk.git
-opam init --bare --no-setup --disable-sandboxing
-opam switch create codehawk-4.12.1 4.12.1 --no-switch
-eval $(opam env --switch=codehawk-4.12.1 --set-switch)
-opam install ocamlfind zarith camlzip extlib lablgtk lablgtk-extras goblint-cil
-cd codehawk/CodeHawk
-./full_make.sh
-```
-
-# Building with shake
-
-### High-level process:
-
-1. Install dependencies
-2. Install the shake build system
-3. Install ocaml
-4. Build CodeHawk
+sudo apt install --no-install-recommends \
```

Comment:
On the latest Ubuntu this also should include `git`, for completeness.

Validated with
```
$ sudo docker run -it --privileged --rm ubuntu bash
$ apt update -y
$ apt install --no-install-recommends build-essential git opam unzip openjdk-21-jdk pkg-config m4 zlib1g-dev libgmp-dev bubblewrap -y
```
then the rest of the instructions.

`--privileged` was necessary for `bubblewrap` to work.